src/api/objects/ObjectAPI.js

```diff
@@ -20,337 +20,387 @@
-        let provider = this.getProvider(domainObject.identifier);
-
-        return provider !== undefined
-            && provider.create !== undefined
-            && provider.update !== undefined;
-    };
-
-    /**
-     * Save this domain object in its current state. EXPERIMENTAL
-     *
-     * @private
-     * @memberof module:openmct.ObjectAPI#
-     * @param {module:openmct.DomainObject} domainObject the domain object to
-     *        save
-     * @returns {Promise} a promise which will resolve when the domain object
-     *          has been saved, or be rejected if it cannot be saved
-     */
-    ObjectAPI.prototype.save = function (domainObject) {
-        let provider = this.getProvider(domainObject.identifier);
-        let savedResolve;
-        let result;
-
-        if (!this.isPersistable(domainObject)) {
-            result = Promise.reject('Object provider does not support saving');
-        } else if (hasAlreadyBeenPersisted(domainObject)) {
-            result = Promise.resolve(true);
+    this.cache[keystring] = objectPromise;
+
+    return objectPromise.then(result => {
+        delete this.cache[keystring];
+
+        return result;
+    });
+};
+
+/**
+ * Will fetch object for the given identifier, returning a version of the object that will automatically keep
+ * itself updated as it is mutated. Before using this function, you should ask yourself whether you really need it.
+ * The platform will provide mutable objects to views automatically if the underlying object can be mutated. The
+ * platform will manage the lifecycle of any mutable objects that it provides. If you use `getMutable` you are
+ * committing to managing that lifecycle yourself. `.destroy` should be called when the object is no longer needed.
+ *
+ * @memberof {module:openmct.ObjectAPI#}
+ * @returns {Promise.<MutableDomainObject|DomainObject>} a promise that will resolve with a MutableDomainObject if
+ * the object can be mutated, or a DomainObject if not.
+ */
+ObjectAPI.prototype.getMutable = function (identifier) {
+    return this.get(identifier).then((object) => {
+        if (this.supportsMutation(object)) {
+            return this._toMutable(object);
         } else {
-            const persistedTime = Date.now();
-            if (domainObject.persisted === undefined) {
-                result = new Promise((resolve) => {
-                    savedResolve = resolve;
-                });
-                domainObject.persisted = persistedTime;
-                provider.create(domainObject).then((response) => {
-                    this.mutate(domainObject, 'persisted', persistedTime);
-                    savedResolve(response);
-                });
-            } else {
-                domainObject.persisted = persistedTime;
-                this.mutate(domainObject, 'persisted', persistedTime);
-                result = provider.update(domainObject);
-            }
+            return object;
```

Comment:
Returning a non-mutable object here seems misleading. We should throw an error if the object does not support mutation.

Reply:
You're right, I'll change it.

Reply:
Oh wait, now I remember 😄 There is a problem with that because there is no way of knowing whether an object is mutable or not based solely on its identifier, you have to fetch it first, so it's a catch-22. 

Let me come back to this. Maybe `isPersistable` is a workable proxy for mutability that doesn't require a fully resolved object?

Reply:
Fixed it using `isPersistable`. `getMutable` will only return mutables, and will error otherwise.